### PR TITLE
Support texture without data - extra Texture::new function

### PR DESCRIPTION
**Pull request by nokola**
Cleaned up code and created new PR

### CODE REVIEW

**Comment by nokola** [comment]
Verified these examples work on Windows:
1. quad
2. offscreen
3. instancing

File: src/graphics/texture.rs [hunk 1]
```diff
@@ -116,7 +137,9 @@ impl Texture {
 
         unsafe {
             glGenTextures(1, &mut texture as *mut _);
-            glActiveTexture(GL_TEXTURE0);
+            if access == TextureAccess::RenderTarget {
+                glActiveTexture(GL_TEXTURE0);
```
Comment (not-fl3) [inline src/graphics/texture.rs:141]: Why only for RenderTarget textures?

Comment (nokola) [reply src/graphics/texture.rs:141]: I am not sure - I copied the logic in `from_data_and_format` below - the only difference between `from_data_and_format` and `new_render_texture` was that `from_data_and_format` does not call glActiveTexture(...).

File: src/graphics/texture.rs [hunk 2]
```diff
@@ -147,45 +173,7 @@ impl Texture {
...
-        );
-
-        let (internal_format, format, pixel_type) = params.format.into();
-
-        unsafe {
-            ctx.cache.store_texture_binding(0);
-
-            let mut texture: GLuint = 0;
-            glGenTextures(1, &mut texture as *mut _);
-            ctx.cache.bind_texture(0, texture);
```
Comment (nokola) [inline src/graphics/texture.rs:162]: just for reference: this is the place where glActiveTexture() was missing - not sure if by accident or intentional

File: src/graphics/texture.rs (hunk 1 above)
Comment (not-fl3) [reply src/graphics/texture.rs:141]: It looks like calling glActiveTexture always is fine. At least I cant see anything broken without that `if` :)

Comment (not-fl3) [reply src/graphics/texture.rs:141]: Oh wait, 
```
    fn bind_texture(&mut self, slot_index: usize, texture: GLuint) {
        unsafe {
            glActiveTexture(GL_TEXTURE0 + slot_index as GLuint);
      ...
```

glActiveTexture is going to be called anyway, glActiveTexture should be just deleted!

**Comment by not-fl3** [comment]
Awesome, thanks!

File: src/graphics/texture.rs (hunk 1 above)
Comment (nokola) [reply src/graphics/texture.rs]: nice! Removed. I still kept the TextureAccess even though unused. Do you think we'll need TextureAccess later? 

I know for example Unity differentiates RenderTexture (render target) vs "regular" Texture2D (GPU read), however in the miniquad code there's currently no implementation difference.
I assume the TextureAccess can be used to speed up _something_ but I'm not sure what - perhaps in the future some optimization with framebuffer object vs texture? (https://community.khronos.org/t/using-textures-as-render-target/66561)

Comment (not-fl3) [reply src/graphics/texture.rs]: Yes, I am pretty sure that there are some possible optimizations with TextureAccess, so its a good change!

**Pull request merged**